pybamm/models/full_battery_models/lithium_ion/spm.py

```diff
@@ -53,7 +53,10 @@ def __init__(self, options=None, name="Single Particle Model", build=True):
 
     def set_porosity_submodel(self):
 
-        self.submodels["porosity"] = pybamm.porosity.Constant(self.param)
+        if self.options["porosity"] is None:
+            self.submodels["porosity"] = pybamm.porosity.Constant(self.param)
+        elif self.options["porosity"] == "variable porosity":
+            self.submodels["porosity"] = pybamm.porosity.SeiFull(self.param)
```

Comment:
For the SPM, given that `j_sei` is uniform in x, using the "SeiFull" porosity submodel creates unnecessarily many states for the porosity (i.e. when discretised, there will be `n` ODEs where `n` is the number of grid points, instead of just one ODE for each domain). You could use the "leading order reaction driven porosity" model instead